cli/cmd/vm_endpoints.go

```diff
@@ -0,0 +1,189 @@
+
+The output will be in the format: %s
+
+You can identify the VM either by its unique ID or by its name.
+
+Note: %s endpoints can only be retrieved from VMs in the "running" state.`, protocol, outputFormat, protocol)
+
+	cmdExample := fmt.Sprintf(`# Get %s endpoint for a specific VM by ID
+replicated vm %s-endpoint aaaaa11
+
+# Get %s endpoint for a specific VM by name
+replicated vm %s-endpoint my-test-vm`, protocol, endpointType, protocol, endpointType)
+
+	cmd := &cobra.Command{
+		Use:               cmdUse,
+		Short:             cmdShort,
+		Long:              cmdLong,
+		Example:           cmdExample,
+		RunE:              r.VMEndpoint,
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: r.completeVMIDsAndNames,
+		Annotations:       map[string]string{"endpointType": endpointType},
+	}
+
+	parent.AddCommand(cmd)
+
+	return cmd
+}
+
+// VMEndpoint handles both the ssh-endpoint and scp-endpoint commands
+func (r *runners) VMEndpoint(cmd *cobra.Command, args []string) error {
+	// Get endpoint type from command annotation
+	endpointType := EndpointTypeSSH // Default fallback
+	if cmd.Annotations != nil {
+		if epType, ok := cmd.Annotations["endpointType"]; ok {
+			endpointType = epType
+		}
+	}
+
+	vmID, err := r.getVMIDFromArg(args[0])
+	if err != nil {
+		return err
+	}
+
+	return r.getVMEndpoint(vmID, endpointType, nil, "")
+}
+
+// getVMEndpoint retrieves and formats VM endpoint with the specified protocol
+// endpointType should be either "ssh" or "scp"
+func (r *runners) getVMEndpoint(vmID, endpointType string, vm *VM, githubUsername string) error {
+	var err error
+
+	// Validate endpoint type
+	if err := validateEndpointType(endpointType); err != nil {
+		return err
+	}
+
+	// Use vm if provided, otherwise fetch from API
+	if vm == nil {
+		vmFromAPI, err := r.kotsAPI.GetVM(vmID)
+		if err != nil {
+			return errors.Wrap(err, "get vm")
+		}
+		vm = &VM{
+			DirectEndpoint: vmFromAPI.DirectSSHEndpoint,
+			DirectPort:     vmFromAPI.DirectSSHPort,
+			ID:             vmFromAPI.ID,
+			Status:         string(vmFromAPI.Status),
+		}
+	}
+
+	if vm.Status != "running" {
+		return errors.Errorf("VM %s is not in running state (current state: %s). %s is only available for running VMs",
+			vm.ID, vm.Status, strings.ToUpper(endpointType))
+	}
+
+	if vm.DirectEndpoint == "" || vm.DirectPort == 0 {
+		return errors.Errorf("VM %s does not have %s endpoint configured", vm.ID, endpointType)
+	}
+
+	// if kotsAPI is not nil and githubUsername is not provided, fetch from API
+	if r.kotsAPI != nil && githubUsername == "" {
+		// Get GitHub username from API
+		githubUsername, err = r.kotsAPI.GetGitHubUsername()
+		if err != nil {
+			return errors.Wrap(err, "get github username")
+		}
+	}
+
+	// Format the endpoint with username if available
+	if githubUsername == "" {
+		return errors.Errorf(`no github account associated with vendor portal user
+Visit https://vendor.replicated.com/account-settings to link your account`)
```

Comment:
```suggestion
Visit the Account Settings page in Vendor Portal to link your account`)
```

Reply:
this pattern is also seen at https://github.com/replicatedhq/replicated/blob/efe7636424e71ed28141b340c8b57b02022a4969/cli/cmd/cluster.go#L13-L14 but I don't suggest changing it in this PR

Reply:
Thank you! I have updated the error message accordingly.